compat/inet_ntop.cc

```diff
@@ -204,8 +200,11 @@ inet_ntop6(const u_char *src, char *dst, size_t size)
     /*
      * Format the result.
      */
-    tp = tmp;
-    for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ); i++) {
+    constexpr int maxbuflen =
```

Comment:
Overall, I am against the unnecessary (here) and dangerous (always) disconnect between `buf` size and expressions that need to know `buf` size. In other words, here, `sizeof(buf)` is safer than a dedicated named constant that is supposed to contain `buf` size. See below for specific suggestions if you decide to keep this constant.

* We should be using `size_t` for object sizes.
* We should be using auto when possible.
* We should be using camelCase.
* This is not a _maximum_ buffer size. It is (constant) buffer size.
* If you add a `buf` size constant, then please use it _everywhere_ instead of `sizeof(buf)`, for consistency sake. Using two ways to refer to the same buffer size increases dangers where parts of this code go out of sync with each other.
* Consexpr is a distracting overkill here IMO. This function cannot be used at compile time, of course.  We should only be using `constexpr` in special cases where we focus on or at least foresee compile-time usage, not everywhere it is usable. FWIW, some folks [share](https://stackoverflow.com/a/6785367) this opinion.

```suggestion
    const auto bufSize =
```

You can use the following test to play with different variations of this declaration. For example, I find the effect of removing `const` interesting: https://godbolt.org/z/Kqjj1xssa

I do not insist on any changes.

Reply:
I'm very okay with using const auto in place of constexpr int and renaming the variable; changing.

I think in this case using temporary variable conveys intent better than using sizeof() twice; in the other case (inet_ntop) the code is much simpler and cleaner, I'd leave as-is there.

I'm not sure I get the philosophical implications constexpr versus const and when to use it, but they are out of scope for this change, where the compiler will very likely optimize all away anyway.

Thanks for your feedback

Reply:
> I'm not sure I get the philosophical implications constexpr versus const and when to use it

I am writing this comment in hope to avoid similar discussions in other PRs: I recommend not using `constexpr` until you know when to use it. If you want to discuss and polish more specific and more general recommendations on `constexpr` use (for future use), I am happy to participate (here or elsewhere)! FWIW, my suggestion for the starting point is "use `constexpr` only where its use is already necessary in the current code or where it is likely to become necessary in future code".  I expect this rule of thumb to result in `constexpr` use in special cases where we focus on or at least foresee compile-time usage of a function or variable with compiler optimizations turned _off_. I suspect `constexpr` is not really focused on performance optimization. It is more about writing clearer code (that also happens to work at compile time).

> philosophical implications constexpr versus const and when to use it ... are out of scope for this change

I agree that they are out of scope as long as the code does not use (and does not have to use) `constexpr`. Thank you for removing it from this PR!